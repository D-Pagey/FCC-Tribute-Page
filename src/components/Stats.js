import React, { Component } from 'react';

class Stats extends Component {
  constructor(props) {
    super(props);
  }

  render() {
    return (
      <div className="details">
        <h4 className="stats-title copy">Stats:</h4>
        <ul className="stats-container">
          <li><span className="stat-name copy">Full Name:</span>
            <span className="stat copy">Chris Omprakash Sharma</span></li>
          <li><span className="stat-name copy">Birthday:</span>
            <span className="stat copy">23rd April 1981</span></li>
          <li><span className="stat-name copy">Nationality:</span>
            <span className="stat copy">American</span></li>
          <li><span className="stat-name copy">Website:</span>
            <span className="stat copy">
              <a href="http://www.chrissharma.com" target="_blank" className="website copy">
                ChrisSharma.com</a></span></li>
        </ul>
      </div>
    )
  }
}

export default Stats;
