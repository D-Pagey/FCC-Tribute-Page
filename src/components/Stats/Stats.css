.details {
  align-self: center;
  margin: 1em 0 0;
  width: 92.5%; }

.stats-title {
  text-decoration: underline; }
  @media screen and (min-width: 1400px) {
    .stats-title {
      font-size: 1.15rem; } }
  @media screen and (min-width: 2000px) {
    .stats-title {
      font-size: 1.5em; } }

.stats-container {
  list-style-type: none; }

.stat-name {
  display: inline-block;
  font-weight: bold;
  width: 6em; }
  @media screen and (min-width: 1400px) {
    .stat-name {
      font-size: 1.15rem; } }
  @media screen and (min-width: 2000px) {
    .stat-name {
      font-size: 1.5em; } }

@media screen and (min-width: 1400px) {
  .stat {
    font-size: 1.15rem; } }
@media screen and (min-width: 2000px) {
  .stat {
    font-size: 1.5em; } }

.website {
  color: #007bff;
  text-decoration: none; }
  @media screen and (min-width: 1400px) {
    .website {
      font-size: 1.15rem; } }
  @media screen and (min-width: 2000px) {
    .website {
      font-size: 1em; } }

@media screen and (min-width: 1400px) {
  .text {
    font-size: 1.15rem; } }
@media screen and (min-width: 2000px) {
  .text {
    font-size: 1.5em; } }
