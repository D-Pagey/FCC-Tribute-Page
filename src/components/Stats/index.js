import React from 'react';

import './index.css';

export default function Stats() {
  return (
    <div className="details">
      <h4 className="stats-title">Stats:</h4>
      <ul className="stats-container">
        <li><span className="stat-name">Full Name:</span>
          <span className="stat">Chris Omprakash Sharma</span></li>
        <li><span className="stat-name">Birthday:</span>
          <span className="stat">23rd April 1981</span></li>
        <li><span className="stat-name">Nationality:</span>
          <span className="stat">American</span></li>
        <li><span className="stat-name">Website:</span>
          <span className="stat">
            <a href="http://www.chrissharma.com" target="_blank" rel='noreferrer noopener'
              className="website">ChrisSharma.com</a></span></li>
      </ul>
    </div>
  )
}
