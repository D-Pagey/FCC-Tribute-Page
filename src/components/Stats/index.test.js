import React from 'react';
import { shallow } from 'enzyme';

import Stats from '.';

it('renders the Stats', () => {
  const wrapper = shallow(
    <Stats />,
  );
  expect(wrapper).toMatchSnapshot();
});
