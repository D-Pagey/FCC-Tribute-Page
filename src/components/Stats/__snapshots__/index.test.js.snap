// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders the Stats 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Stats />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <h4
          className="stats-title"
>
          Stats:
</h4>,
        <ul
          className="stats-container"
>
          <li>
                    <span
                              className="stat-name"
                    >
                              Full Name:
                    </span>
                    <span
                              className="stat"
                    >
                              Chris Omprakash Sharma
                    </span>
          </li>
          <li>
                    <span
                              className="stat-name"
                    >
                              Birthday:
                    </span>
                    <span
                              className="stat"
                    >
                              23rd April 1981
                    </span>
          </li>
          <li>
                    <span
                              className="stat-name"
                    >
                              Nationality:
                    </span>
                    <span
                              className="stat"
                    >
                              American
                    </span>
          </li>
          <li>
                    <span
                              className="stat-name"
                    >
                              Website:
                    </span>
                    <span
                              className="stat"
                    >
                              <a
                                        className="website"
                                        href="http://www.chrissharma.com"
                                        rel="noreferrer noopener"
                                        target="_blank"
                              >
                                        ChrisSharma.com
                              </a>
                    </span>
          </li>
</ul>,
      ],
      "className": "details",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Stats:",
          "className": "stats-title",
        },
        "ref": null,
        "rendered": "Stats:",
        "type": "h4",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <li>
              <span
                            className="stat-name"
              >
                            Full Name:
              </span>
              <span
                            className="stat"
              >
                            Chris Omprakash Sharma
              </span>
</li>,
            <li>
              <span
                            className="stat-name"
              >
                            Birthday:
              </span>
              <span
                            className="stat"
              >
                            23rd April 1981
              </span>
</li>,
            <li>
              <span
                            className="stat-name"
              >
                            Nationality:
              </span>
              <span
                            className="stat"
              >
                            American
              </span>
</li>,
            <li>
              <span
                            className="stat-name"
              >
                            Website:
              </span>
              <span
                            className="stat"
              >
                            <a
                                          className="website"
                                          href="http://www.chrissharma.com"
                                          rel="noreferrer noopener"
                                          target="_blank"
                            >
                                          ChrisSharma.com
                            </a>
              </span>
</li>,
          ],
          "className": "stats-container",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                <span
                  className="stat-name"
>
                  Full Name:
</span>,
                <span
                  className="stat"
>
                  Chris Omprakash Sharma
</span>,
              ],
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "Full Name:",
                  "className": "stat-name",
                },
                "ref": null,
                "rendered": "Full Name:",
                "type": "span",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "Chris Omprakash Sharma",
                  "className": "stat",
                },
                "ref": null,
                "rendered": "Chris Omprakash Sharma",
                "type": "span",
              },
            ],
            "type": "li",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                <span
                  className="stat-name"
>
                  Birthday:
</span>,
                <span
                  className="stat"
>
                  23rd April 1981
</span>,
              ],
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "Birthday:",
                  "className": "stat-name",
                },
                "ref": null,
                "rendered": "Birthday:",
                "type": "span",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "23rd April 1981",
                  "className": "stat",
                },
                "ref": null,
                "rendered": "23rd April 1981",
                "type": "span",
              },
            ],
            "type": "li",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                <span
                  className="stat-name"
>
                  Nationality:
</span>,
                <span
                  className="stat"
>
                  American
</span>,
              ],
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "Nationality:",
                  "className": "stat-name",
                },
                "ref": null,
                "rendered": "Nationality:",
                "type": "span",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "American",
                  "className": "stat",
                },
                "ref": null,
                "rendered": "American",
                "type": "span",
              },
            ],
            "type": "li",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                <span
                  className="stat-name"
>
                  Website:
</span>,
                <span
                  className="stat"
>
                  <a
                                    className="website"
                                    href="http://www.chrissharma.com"
                                    rel="noreferrer noopener"
                                    target="_blank"
                  >
                                    ChrisSharma.com
                  </a>
</span>,
              ],
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "Website:",
                  "className": "stat-name",
                },
                "ref": null,
                "rendered": "Website:",
                "type": "span",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": <a
                    className="website"
                    href="http://www.chrissharma.com"
                    rel="noreferrer noopener"
                    target="_blank"
>
                    ChrisSharma.com
</a>,
                  "className": "stat",
                },
                "ref": null,
                "rendered": Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": "ChrisSharma.com",
                    "className": "website",
                    "href": "http://www.chrissharma.com",
                    "rel": "noreferrer noopener",
                    "target": "_blank",
                  },
                  "ref": null,
                  "rendered": "ChrisSharma.com",
                  "type": "a",
                },
                "type": "span",
              },
            ],
            "type": "li",
          },
        ],
        "type": "ul",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <h4
            className="stats-title"
>
            Stats:
</h4>,
          <ul
            className="stats-container"
>
            <li>
                        <span
                                    className="stat-name"
                        >
                                    Full Name:
                        </span>
                        <span
                                    className="stat"
                        >
                                    Chris Omprakash Sharma
                        </span>
            </li>
            <li>
                        <span
                                    className="stat-name"
                        >
                                    Birthday:
                        </span>
                        <span
                                    className="stat"
                        >
                                    23rd April 1981
                        </span>
            </li>
            <li>
                        <span
                                    className="stat-name"
                        >
                                    Nationality:
                        </span>
                        <span
                                    className="stat"
                        >
                                    American
                        </span>
            </li>
            <li>
                        <span
                                    className="stat-name"
                        >
                                    Website:
                        </span>
                        <span
                                    className="stat"
                        >
                                    <a
                                                className="website"
                                                href="http://www.chrissharma.com"
                                                rel="noreferrer noopener"
                                                target="_blank"
                                    >
                                                ChrisSharma.com
                                    </a>
                        </span>
            </li>
</ul>,
        ],
        "className": "details",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Stats:",
            "className": "stats-title",
          },
          "ref": null,
          "rendered": "Stats:",
          "type": "h4",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <li>
                <span
                                className="stat-name"
                >
                                Full Name:
                </span>
                <span
                                className="stat"
                >
                                Chris Omprakash Sharma
                </span>
</li>,
              <li>
                <span
                                className="stat-name"
                >
                                Birthday:
                </span>
                <span
                                className="stat"
                >
                                23rd April 1981
                </span>
</li>,
              <li>
                <span
                                className="stat-name"
                >
                                Nationality:
                </span>
                <span
                                className="stat"
                >
                                American
                </span>
</li>,
              <li>
                <span
                                className="stat-name"
                >
                                Website:
                </span>
                <span
                                className="stat"
                >
                                <a
                                                className="website"
                                                href="http://www.chrissharma.com"
                                                rel="noreferrer noopener"
                                                target="_blank"
                                >
                                                ChrisSharma.com
                                </a>
                </span>
</li>,
            ],
            "className": "stats-container",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  <span
                    className="stat-name"
>
                    Full Name:
</span>,
                  <span
                    className="stat"
>
                    Chris Omprakash Sharma
</span>,
                ],
              },
              "ref": null,
              "rendered": Array [
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": "Full Name:",
                    "className": "stat-name",
                  },
                  "ref": null,
                  "rendered": "Full Name:",
                  "type": "span",
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": "Chris Omprakash Sharma",
                    "className": "stat",
                  },
                  "ref": null,
                  "rendered": "Chris Omprakash Sharma",
                  "type": "span",
                },
              ],
              "type": "li",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  <span
                    className="stat-name"
>
                    Birthday:
</span>,
                  <span
                    className="stat"
>
                    23rd April 1981
</span>,
                ],
              },
              "ref": null,
              "rendered": Array [
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": "Birthday:",
                    "className": "stat-name",
                  },
                  "ref": null,
                  "rendered": "Birthday:",
                  "type": "span",
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": "23rd April 1981",
                    "className": "stat",
                  },
                  "ref": null,
                  "rendered": "23rd April 1981",
                  "type": "span",
                },
              ],
              "type": "li",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  <span
                    className="stat-name"
>
                    Nationality:
</span>,
                  <span
                    className="stat"
>
                    American
</span>,
                ],
              },
              "ref": null,
              "rendered": Array [
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": "Nationality:",
                    "className": "stat-name",
                  },
                  "ref": null,
                  "rendered": "Nationality:",
                  "type": "span",
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": "American",
                    "className": "stat",
                  },
                  "ref": null,
                  "rendered": "American",
                  "type": "span",
                },
              ],
              "type": "li",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  <span
                    className="stat-name"
>
                    Website:
</span>,
                  <span
                    className="stat"
>
                    <a
                                        className="website"
                                        href="http://www.chrissharma.com"
                                        rel="noreferrer noopener"
                                        target="_blank"
                    >
                                        ChrisSharma.com
                    </a>
</span>,
                ],
              },
              "ref": null,
              "rendered": Array [
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": "Website:",
                    "className": "stat-name",
                  },
                  "ref": null,
                  "rendered": "Website:",
                  "type": "span",
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": <a
                      className="website"
                      href="http://www.chrissharma.com"
                      rel="noreferrer noopener"
                      target="_blank"
>
                      ChrisSharma.com
</a>,
                    "className": "stat",
                  },
                  "ref": null,
                  "rendered": Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "host",
                    "props": Object {
                      "children": "ChrisSharma.com",
                      "className": "website",
                      "href": "http://www.chrissharma.com",
                      "rel": "noreferrer noopener",
                      "target": "_blank",
                    },
                    "ref": null,
                    "rendered": "ChrisSharma.com",
                    "type": "a",
                  },
                  "type": "span",
                },
              ],
              "type": "li",
            },
          ],
          "type": "ul",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
